pkg/controllers/managementuser/rbac/crtb_handler.go

```diff
@@ -17,123 +18,223 @@ import (
 	return &crtbLifecycle{
-		m:          m,
-		rtLister:   management.Management.RoleTemplates("").Controller().Lister(),
-		crbLister:  m.workload.RBAC.ClusterRoleBindings("").Controller().Lister(),
-		crbClient:  m.workload.RBAC.ClusterRoleBindings(""),
-		crtbClient: management.Management.ClusterRoleTemplateBindings(""),
+		m:           m,
+		rtLister:    management.Management.RoleTemplates("").Controller().Lister(),
+		crbLister:   m.workload.RBAC.ClusterRoleBindings("").Controller().Lister(),
+		crbClient:   m.workload.RBAC.ClusterRoleBindings(""),
+		crtbClient:  management.Management.ClusterRoleTemplateBindings(""),
+		crtbClientM: management.WithAgent("rbac-handler-base").Wrangler.Mgmt.ClusterRoleTemplateBinding(),
 	}
 }
 
+// Local interface abstracting the mgmtconv3.ClusterRoleTemplateBindingController down to
+// necessities. The testsuite then provides a local mock implementation for itself.
+type clusterRoleTemplateBindingController interface {
+	UpdateStatus(*v3.ClusterRoleTemplateBinding) (*v3.ClusterRoleTemplateBinding, error)
+}
+
 type crtbLifecycle struct {
-	m          managerInterface
-	rtLister   v3.RoleTemplateLister
-	crbLister  typesrbacv1.ClusterRoleBindingLister
-	crbClient  typesrbacv1.ClusterRoleBindingInterface
-	crtbClient v3.ClusterRoleTemplateBindingInterface
+	m           managerInterface
+	rtLister    v3.RoleTemplateLister
+	crbLister   typesrbacv1.ClusterRoleBindingLister
+	crbClient   typesrbacv1.ClusterRoleBindingInterface
+	crtbClient  v3.ClusterRoleTemplateBindingInterface
+	crtbClientM clusterRoleTemplateBindingController
 }
 
 func (c *crtbLifecycle) Create(obj *v3.ClusterRoleTemplateBinding) (runtime.Object, error) {
-	err := c.syncCRTB(obj)
-	return obj, err
+	// ObjectMeta.Generation does not get updated when the Status is updated.
+	// If only the status has been updated and we have finished updating the status
+	// (status.Summary != "InProgress") we don't need to perform a reconcile as nothing has
+	// changed.
+	if obj.Status.ObservedGeneration == obj.ObjectMeta.Generation &&
+		obj.Status.Summary != SummaryInProgress {
+		return obj, nil
+	}
+	returnError := errors.Join(
+		c.setCRTBAsInProgress(obj),
+		c.syncCRTB(obj),
+		c.setCRTBAsCompleted(obj),
+	)
+	return obj, returnError
 }
 
 func (c *crtbLifecycle) Updated(obj *v3.ClusterRoleTemplateBinding) (runtime.Object, error) {
-	if err := c.reconcileCRTBUserClusterLabels(obj); err != nil {
-		return obj, err
+	// ObjectMeta.Generation does not get updated when the Status is updated.
+	// If only the status has been updated and we have finished updating the status
+	// (status.Summary != "InProgress") we don't need to perform a reconcile as nothing has
+	// changed.
+	if obj.Status.ObservedGeneration == obj.ObjectMeta.Generation &&
+		obj.Status.Summary != SummaryInProgress {
+		return obj, nil
 	}
-	err := c.syncCRTB(obj)
-	return obj, err
+	returnError := errors.Join(
```

Comment:
I don't think we _need_ to have the labels on the CRTB before `syncCRTB` happens, but for the sake of avoiding some unforeseen regression, I think they should be kept separate in this PR

Reply:
Will do.

Reply:
done with commit 4eb67542181b1a6ad0679c3f95e4d6e0406af4ff